File: mvpLibrary/src/main/java/cn/cbsd/mvplibrary/mvp/IView.java
package cn.cbsd.mvplibrary.mvp;

import android.os.Bundle;
import android.view.View;

/**
 * Created by wanglei on 2016/12/29.
 */

public interface IView<P> {

    void bindUI(View rootView);

    void bindEvent();

    int getLayoutId();

    void initData(Bundle savedInstanceState);

    int getOptionsMenuId();

    boolean useEventBus();

    P newP();

    void showLoading();
    void showLoading(String msg);

    void hideLoading();

    boolean useUiState();

}

File: mvpLibrary/src/main/java/cn/cbsd/mvplibrary/mvp/IView.kt
package cn.cbsd.mvplibrary.mvp

import android.os.Bundle
import android.view.View

/**
 * Created by wanglei on 2016/12/29.
 */

interface IView {

    val layoutId: Int

    val optionsMenuId: Int

    fun bindUI(rootView: View?)

    fun bindEvent()

    fun initData(savedInstanceState: Bundle?)

    fun useEventBus(): Boolean

    fun showLoading()

    fun showLoading(msg: String?)

    fun hideLoading()

    fun useDefaultUiState(): Boolean

    /**
     * 没用，只是用来兼容之前的版本
     */
    fun newP(): Any?
}
